compiler/src/lnntoamm/opcodes.ts

```diff
@@ -0,0 +1,66 @@
+import Event from './Event';
+import { OpcodeFn } from './Fn';
+import Scope from "./Scope";
+import Type from './Types';
+
+let __opcodes: Scope = null;
+
+const opcodes = (): Scope => {
+  if (__opcodes === null) load();
+  return __opcodes;
+};
+
+export default opcodes;
+
+const load = (): void => {
+  __opcodes = new Scope();
+
+  Object.entries({
+    void: [],
+    int8: [],
+    int16: [],
+    int32: [],
+    int64: [],
+    float32: [],
+    float64: [],
+    bool: [],
+    string: [],
+  }).forEach(([name, generics]: [string, string[]]) => {
```

Comment:
Am I reading this correctly? Are you destructuring one argument into two and typing them in the arguments declaration of this anonymous function?

Reply:
`Object.entries` returns a list of tuples, so in this case it's `[string, any[]][]`. The only the argument's type declaration is doing is specifying that the value of each key isn't just `any[]`, it's actually `string[]`. Otherwise, I'm just destructuring the tuple `[key, value]`

Reply:
Got it. I didn't notice the `entries` usage.